core/src/main/scala/org/bitcoins/core/wallet/utxo/AddressTag.scala

```diff
@@ -0,0 +1,14 @@
+package org.bitcoins.core.wallet.utxo
+
+case class AddressTagType(name: String)
+
+case class AddressTag(tag: String, tagTypeOpt: Option[AddressTagType])
+
+trait AddressTagFactory[TagType <: AddressTagType] {
```

Comment:
Not sure this makes sense considering there is only one type of `AddressTagType` as the class cannot be extended.